src/core/include/lattice/hal/default/dcrtpoly-impl.h

```diff
@@ -808,173 +778,83 @@ void DCRTPolyImpl<VecType>::ModReduce(const NativeInteger& t, const std::vector<
     }
 }
 
-/* methods to access individual members of the DCRTPolyImpl. Result is
- * Interpolated value at that point.  Note this is a very costly compute
- * intensive operation meant basically for debugging code.
- */
-template <typename VecType>
-typename VecType::Integer& DCRTPolyImpl<VecType>::at(usint i) {
-    if (0 == m_vectors.size())
-        OPENFHE_THROW(math_error, "No values in DCRTPolyImpl");
-    if (i >= m_vectors.size())
-        OPENFHE_THROW(math_error, "out of range in  DCRTPolyImpl.at()");
-    return CRTInterpolateIndex(i)[i];
-}
-
-template <typename VecType>
-const typename VecType::Integer& DCRTPolyImpl<VecType>::at(usint i) const {
-    if (0 == m_vectors.size())
-        OPENFHE_THROW(math_error, "No values in DCRTPolyImpl");
-    if (i >= m_vectors.size())
-        OPENFHE_THROW(math_error, "out of range in  DCRTPolyImpl.at()");
-    return CRTInterpolateIndex(i)[i];
-}
-
 /*
- * This method applies the Chinese Remainder Interpolation on an DCRTPolyImpl
- * and produces an Poly How the Algorithm works: Consider the DCRTPolyImpl as
- * a 2-dimensional matrix M, with dimension ringDimension * Number of Towers.
- * For brevity , lets say this is r * t Let qt denote the bigModulus (all the
- * towers' moduli multiplied together) and qi denote the modulus of a
- * particular tower. Let V be a BigVector of size tower (tower size). Each
- * coefficient of V is calculated as follows: for every r calculate: V[j]=
- * {Sigma(i = 0 --> t-1) ValueOf M(r,i) * qt/qi *[ (qt/qi)^(-1) mod qi ]}mod
- * qt
- *
- * Once we have the V values, we construct an Poly from V, use qt as it's
- * modulus, and calculate a root of unity for parameter selection of the Poly.
+ * This method applies Chinese Remainder Interpolation on a DCRTPoly
+ * How the Algorithm works:
+ * View the DCRTPoly as a (t = Number of Towers) x (r = ring Dimension) Matrix M.
+ * Let qt denote the bigModulus (product of each tower moduli), qi denote the
+ * modulus of a particular tower, and V be a BigVector of length r.
+ * For j = 0 --> r-1, calculate
+ * V[j] = Sigma(i = 0 --> t-1) { M(i,j) * qt/qi * [(qt/qi)^(-1) mod qi] } mod qt
  */
 template <typename VecType>
 typename DCRTPolyImpl<VecType>::PolyLargeType DCRTPolyImpl<VecType>::CRTInterpolate() const {
-    usint ringDimension = m_params->GetRingDimension();
-    usint nTowers       = m_vectors.size();
-
-    Integer bigModulus(m_params->GetModulus());  // qT
-
-    // this is the resulting vector of coefficients
-    VecType coefficients(ringDimension, bigModulus);
+    if (m_format == Format::EVALUATION)
```

Comment:
could it be replaced with "if (m_format != Format::COEFFICIENT)"?

Reply:
will update in PR for issue 458